pkg/api/defaults.go

```diff
@@ -917,6 +921,25 @@ func (p *Properties) setTelemetryProfileDefaults() {
 	}
 }
 
+func (cs *ContainerService) setConnectedClusterProfileDefaults() {
+	p := cs.Properties
+	if p.ConnectedClusterProfile == nil {
+		p.ConnectedClusterProfile = &ConnectedClusterProfile{}
+	}
+	if p.ConnectedClusterProfile.Location == "" {
+		p.ConnectedClusterProfile.Location = cs.Location
+	}
+	if p.ConnectedClusterProfile.ClientID == "" {
+		p.ConnectedClusterProfile.ClientID = p.ServicePrincipalProfile.ClientID
+	}
+	if p.ConnectedClusterProfile.ClientSecret == "" {
+		p.ConnectedClusterProfile.ClientSecret = p.ServicePrincipalProfile.Secret
+	}
+	// p.ConnectedClusterProfile.ClusterName = default to resource group name?
```

Comment:
@jackfrancis is there a way for `defaults.go` to access `resource-group` and `subscriptionID` other than passing them to method `SetPropertiesDefaults`? would it be acceptable to add these two fields to `PropertiesDefaultsParams`?